packages/fixie-sdk/src/types.ts

```diff
@@ -0,0 +1,48 @@
+export interface MessageBase {
+  metadata?: Record<string, string | number | boolean | object | null>;
+}
+
+export interface FunctionCallMessage extends MessageBase {
+  kind: 'functionCall';
+  name: string;
+  args: Record<string, string | number | boolean | null>;
+}
+
+export interface FunctionResponseMessage extends MessageBase {
+  kind: 'functionResponse';
+  name: string;
+  response: string;
+  failed: boolean;
+}
+
+export interface TextMessage extends MessageBase {
+  kind: 'text';
+  content: string;
+}
+
+export type Message = FunctionCallMessage | FunctionResponseMessage | TextMessage;
+
+export interface ConversationTurn {
```

Comment:
Did we decide not to pursue turns being a full user->assistant exchange?

Reply:
I ended up modeling this by pointing the assistant turn back to the user turn via `reply_to_turn_id` -- the main thing that was gained by that was that regenerations could have new turn IDs while pointing to the original/unchanged user turn.